src/servers/kfserving_http_server.cc

```diff
@@ -0,0 +1,949 @@
+        return TRTSERVER_ErrorNew(
+            TRTSERVER_ERROR_INTERNAL,
+            std::string(
+                "reserved " + std::to_string(output_iovec.iov_len) +
+                " bytes in output tensor buffer, need " +
+                std::to_string(byte_size))
+                .c_str());
+      }
+
+      output_iovec.iov_len = byte_size;
+      *buffer = output_iovec.iov_base;
+
+      // Immediately commit the buffer space. We are relying on evbuffer
+      // not to relocate this space. Because we request a contiguous
+      // chunk every time (above by allowing only a single entry in
+      // output_iovec), this seems to be a valid assumption.
+      if (evbuffer_commit_space(evhttp_buffer, &output_iovec, 1) != 0) {
+        *buffer = nullptr;
+        return TRTSERVER_ErrorNew(
+            TRTSERVER_ERROR_INTERNAL,
+            "failed to commit output tensors to output buffer");
+      }
+    }
+  }
+
+  LOG_VERBOSE(1) << "KFServing HTTP allocation: '" << tensor_name
+                 << "', size: " << byte_size << ", addr: " << *buffer;
+
+  return nullptr;  // Success
+}
+
+TRTSERVER_Error*
+KFServingHTTPAPIServer::ResponseRelease(
+    TRTSERVER_ResponseAllocator* allocator, void* buffer, void* buffer_userp,
+    size_t byte_size, TRTSERVER_Memory_Type memory_type, int64_t memory_type_id)
+{
+  LOG_VERBOSE(1) << "KFServing HTTP release: "
+                 << "size " << byte_size << ", addr " << buffer;
+
+  // Don't do anything when releasing a buffer since ResponseAlloc
+  // wrote directly into the response ebvuffer.
+  return nullptr;  // Success
+}
+
+void
+KFServingHTTPAPIServer::Handle(evhtp_request_t* req)
+{
+  LOG_VERBOSE(1) << "KFServing HTTP request: " << req->method << " "
+                 << req->uri->path->full;
+
+  std::string model_name, rest;
+  if (RE2::FullMatch(
+          std::string(req->uri->path->full), api_regex_, &model_name, &rest)) {
+    // TODO Try to verify that model name is valid
+    LOG_VERBOSE(1) << "model_name: " << model_name << ", rest: " << rest;
+
+    // status
+    if (rest == "/metadata") {
+      HandleStatus(req, model_name);
+      return;
+    }
+    // health
+    if ((rest == "") && (model_name != "metadata")) {
+      HandleHealth(req, model_name);
+      return;
+    }
+    // infer
+    if (rest == ":predict") {
+      HandleInfer(req, model_name);
+      return;
+    }
+  }
+
+  LOG_VERBOSE(1) << "KFServing HTTP error: " << req->method << " "
+                 << req->uri->path->full << " - "
+                 << static_cast<int>(EVHTP_RES_BADREQ);
+  evhtp_send_reply(req, EVHTP_RES_BADREQ);
+}
+
+void
+KFServingHTTPAPIServer::HandleHealth(
+    evhtp_request_t* req, const std::string& model_name)
+{
+  if (req->method != htp_method_GET) {
+    evhtp_send_reply(req, EVHTP_RES_METHNALLOWED);
+    return;
+  }
+
+  if (model_name.empty()) {
+    evhtp_send_reply(req, EVHTP_RES_BADREQ);
+    return;
+  }
+
+  bool ready = true;
```

Comment:
Shouldn't you start with 'ready' false and then set it to true below only if everything works correctly and the status is MODEL_READY?

Reply:
Yes, fixed. READY is returned if all versions of the model have the MODEL_READY status